Drv/TcpServer/test/ut/TcpServerTester.cpp

```diff
@@ -84,23 +82,50 @@ void TcpServerTester ::test_with_loop(U32 iterations, bool recv_thread) {
             if (recv_thread) {
                 m_spinner = false;
                 m_data_buffer.setSize(sizeof(m_data_storage));
-                client.send(m_data_buffer.getData(), m_data_buffer.getSize());
+                status2 = client.send(client_fd, m_data_buffer.getData(), m_data_buffer.getSize());
+                EXPECT_EQ(status2, Drv::SOCK_SUCCESS);
+                //from_deallocate_handler(0, m_data_buffer);
                 while (not m_spinner) {}
             }
         }
-        client.close(); // Client must be closed first or the server risks binding to an existing address
+
         // Properly stop the client on the last iteration
         if ((1 + i) == iterations && recv_thread) {
-            this->component.shutdown();
+            // TODO: for some reason you have to stop the server receive thread before closing the client or the server will erroneously receive
```

Comment:
TODO?  Is the server erroring, or is releasing the last empty packet?

Reply:
Thread issue :(